clang/lib/Sema/SemaInit.cpp

```diff
@@ -10636,6 +10889,113 @@ QualType Sema::DeduceTemplateSpecializationFromInitializer(
   // clear on this, but they're not found by name so access does not apply.
   Guides.suppressDiagnostics();
 
+  SmallVector<DeclAccessPair> GuidesCandidates;
+  if (AliasTemplate) {
```

Comment:
I was wondering if  we need to separate the code that tweaks the candidate set into another function.

Reply:
Yes, and this is not the proper place to construct deduction guides for alias templates (they should live in `DeclareImplicitDeductionGuides`).